src/hotspot/cpu/x86/interp_masm_x86.cpp

```diff
@@ -2065,3 +2065,17 @@ void InterpreterMacroAssembler::notify_method_exit(
     pop(state);
   }
 }
+
+void InterpreterMacroAssembler::load_resolved_indy_entry(Register cache, Register index) {
+  // Get index out of bytecode pointer, get_cache_entry_pointer_at_bcp
+  get_cache_index_at_bcp(index, 1, sizeof(u4));
+  // Get address of invokedynamic array
+  movptr(cache, Address(rbp, frame::interpreter_frame_cache_offset * wordSize));
+  movptr(cache, Address(cache, in_bytes(ConstantPoolCache::invokedynamic_entries_offset())));
+  if (is_power_of_2(sizeof(ResolvedIndyEntry))) {
```

Comment:
This was a good suggestion but I wonder if we should assert ResolvedIndyEntry is a power of 2 so we know if we change the size and make it go the slower path?  Or is 32 bit not a power of two and we need this?

Reply:
Currently the structure is a power of two on 64 bits but this is not the case on 32 bit systems.